src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -177,11 +177,18 @@ internal sealed class SourcePropertySymbol : PropertySymbol, IAttributeTargetSym
             {
                 var hasGetSyntax = getSyntax != null;
                 _isAutoProperty = notRegularProperty && hasGetSyntax;
-                bool isReadOnly = hasGetSyntax && setSyntax == null;
+                bool isGetterOnly = hasGetSyntax && setSyntax == null;
 
-                if (_isAutoProperty && !isReadOnly && !IsStatic && ContainingType.IsReadOnly)
+                if (_isAutoProperty && !IsStatic)
                 {
-                    diagnostics.Add(ErrorCode.ERR_AutoPropsInRoStruct, location);
+                    if (!isGetterOnly && ContainingType.IsReadOnly)
+                    {
+                        diagnostics.Add(ErrorCode.ERR_AutoPropsInRoStruct, location);
+                    }
+                    else if (HasReadOnlyModifier)
```

Comment:
Shouldn't this be conditioned on `!isGetterOnly`. A `readonly` modifier should be allowed on a `get` only property. It's just an explicit default.

Reply:
This implementation forbids making any auto-property or accessor `readonly`. Sounds like we don't have consensus on this design quite yet. Will revisit this once we do. @agocke

Reply:
I feel like adding `readonly` to an auto-property is logically no different than adding `readonly` to any member of a `readonly struct`. Not sure how we could allow one without the other.

Additionally it's the act of making implicit behavior explicit. C# generally speaking allows this and in some cases, like `static class`, actually requires it.

Reply:
Done.